android-components/components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -98,12 +104,23 @@ class ContileTopSitesProvider(
             if (response.isSuccess) {
                 val responseBody = response.body.string(Charsets.UTF_8)
 
+                if (response.status == Response.NOT_MODIFIED) {
+                    diskCacheMaxAge = null
+                    getCacheFile().delete()
+                    return listOf()
+                }
+
                 return try {
-                    JSONObject(responseBody).getTopSites().also {
-                        if (maxCacheAgeInMinutes > 0) {
-                            writeToDiskCache(responseBody)
-                        }
-                    }
+                    writeToDiskCache(
+                        if (maxCacheAgeInMinutes == -1L) {
+                            response.headers.computeValidFor() * MILLIS_PER_SEC
+                        } else {
+                            maxCacheAgeInMinutes * MINUTE_IN_MS
```

Comment:
Looking at this, I'm wondering if it would be worth changing our local cache ages to also be in seconds to better match the server and keep the number of types of conversion lower.